sound/soc/sof/pcm.c

```diff
@@ -779,6 +779,15 @@ static int sof_pcm_probe(struct snd_soc_component *component)
 
 static void sof_pcm_remove(struct snd_soc_component *component)
 {
```

Comment:
@ranj063 can you point out what is the current approach and why do we want to change it?

Reply:
@dbaluta what we do today is set up all the widgets and routes that are in topology for all the pipelines. So even if only one pipeline is active, we'd still have the memory allocated for all other pipelines that are not current active as well. This PR changes that so that when we open a PCM only the widgets and connections required for that pipeline will be set up and these will be freed as soon as the PCM is closed.